src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -259,4 +260,177 @@ public void systemEndsPatientVisit() {
 		assertNull(visitsDashboardPage.getActiveVisit());
 		dashboardPage = visitsDashboardPage.goToPatientDashboard();
 	}
+
+	// User story: Adding New Allergy
+	@When("user clicks on Allergies link from Patient dashboard")
+	public void launchManageAllergies() {
+		allergyPage = (AllergyPage) dashboardPage.clickOnAllergiesWidgetLink()
+				.waitForPage();
+	}
+
+	@Then("system loads Allergies table page")
+	public void systemLoadsManageAllergies() {
+				getElement(patientHeaderId).getText();
+		assertTrue(textExists("Allergies"));
+	}
+
+	@When("a user clicks Add New Allergy button")
+	public void addNewAllergy() {
+		allergyPage.clickOnAddNewAllergy();
+	}
+
+	@Then("the system loads Add New Allergy dashboard")
+	public void systemLoadsAddNewAllergy() {
+		assertTrue(textExists("Add New Allergy"));
+	}
+
+	@And("the user selects a new Allergy")
+	public void selectNewAllergy() {
+		allergyPage.clickOn(By.id("allergen-73667-label"));// Codeine
+	}
+
+	@And("the user clicks on the Save button")
+	public void saveAllergy() {
+		allergyPage.clickOn(By.id("addAllergyBtn"));
```

Comment:
Refer to my comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/137/files#r718383180) !